docs/architecture/telemetry.md

```diff
@@ -0,0 +1,289 @@
+# Telemetry
+## Overview
+
+In order to provide consistent capture of telemetry data for the components of Chef Workstation,
+we will extend Chef Workstation App to provide a simple telemetry API over REST.
+
+
+## Constraints
+
+* user opt-in preferences must be respected in all cases.
+* transient opt-out via environment variable `CHEF_TELEMETRY_OPT_OUT` must be respected to
+prevent payloads from being sent upstream.
+
+## Configuration
+
+The telemetry service in CWA will listen on configurable local port 21000. Configuration will be in
+`~/.chef-workstation/config.toml`:
+
+```
+[chef-workstation-tray]
+local_port=21000
+```
+
+## Service
+
+This service will initially be implemented as a hidden Electron window; it may be moved to a non-UI-bound component
+in the future.
+
+### Inbound Requests
+
+When a payload is received by CWA Telemetry Service via a POST to the `/telemetry` endpoint,
+the service will start asynchronous preparation of the final payload and return an HTTP 201 immediately.
+The caller will not receive success/failure notification of the payload's final upstream disposition.
+
+If the payload field `local_opt_out` contains any value other than nil, or the config
```

Comment:
Do we want to call out here that `local_opt_out` will be defaulted by the client to the `ENV['CHEF_TELEMETRY_OPT_OUT']` variable? Or is that part of the implementation details?

Reply:
I did capture that in the card for the client-side changes; it didn't seem relevant to the design for server-side though.